src/lib.ml

```diff
@@ -1093,17 +1112,25 @@ module DB = struct
       resolve_user_deps t (Lib_info.Deps.of_lib_deps deps) ~pps
         ~stack:Dep_stack.empty ~allow_private_deps:true
     in
-    let requires =
+    let requires_link =
       res
       >>=
       closure_with_overlap_checks (Option.some_if (not allow_overlaps) t)
         ~linking:true
       |> Result.map_error ~f:(fun e ->
         Dep_path.prepend_exn e (Executables exes))
     in
+    let requires_compile = lazy (
+      res >>= fun direct_requires ->
```

Comment:
Same question here

Reply:
I should factor this code. Preserving the order from the closure as done previously.